packages/foam-vscode/src/features/dataviz.ts

```diff
@@ -0,0 +1,111 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import { FoamFeature } from "../types";
+import { Foam } from "foam-core";
+import { TextDecoder } from "util";
+
+const feature: FoamFeature = {
+  activate: (context: vscode.ExtensionContext, foamPromise: Promise<Foam>) => {
+    vscode.commands.registerCommand("foam-vscode.show-graph", async () => {
+      const foam = await foamPromise;
+      const panel = vscode.window.createWebviewPanel(
+        "foam-graph",
+        "Foam Graph",
+        vscode.ViewColumn.Two,
+        {
+          enableScripts: true,
+          retainContextWhenHidden: true
+        }
+      );
+
+      const graph = {
+        nodes: [],
+        edges: []
+      };
+
+      foam.notes.getNotes().forEach(n => {
+        graph.nodes.push({
+          id: n.id,
+          type: "note",
+          uri: n.source.uri,
+          title: path.basename(n.source.uri),
```

Comment:
Markdown links grabs the title of the file. Should we be maintaining that functionality? Or is this more a proof of concept PR that we can refine later?

Reply:
that's a great point, and a wider one in that regard.

In the context of this line of code, it should really be `title: n.title`.

I had a conversation with @jevakallio some time ago about how to extract the `title` from filenames/headings/yaml.
Long story short (happy to elaborate offline), the idea is to go back to simplicity, where filenames are what we can rely on, and use those. We could make this more configurable, the wider point is that this needs to be reconsidered.

Let me know if this makes sense :)

Reply:
more context on this: https://github.com/foambubble/foam/blob/master/docs/foam-file-format.md